File: backend/goalPostObject.go
package main

type goalPostObject struct {
	GoalId int64 `json:"goalId"`
	// Unix epoch seconds
	DateTime int64 `json:"dateTime"`
	// HTML
	Text                 string `json:"text"`
	IsAutoTranslated     bool   `json:"isAutoTranslated"`
	IsTranslationPending bool   `json:"isTranslationPending"`
	LanguageName         string `json:"languageName"`
	LanguageTag          string `json:"languageTag"`
	IsPublic             bool   `json:"isPublic"`
}

File: frontend/src/typescript/personal-goals/goalPostObject.ts
import { DateTime } from 'luxon';

export class GoalPostObject {
	constructor(
		public goalId: number = 0,
		/** Unix epoch seconds */
		public dateTime: number = 0,
		/** HTML */
		public text: string = '',
		public isAutoTranslated: boolean = false,
		public isTranslationPending: boolean = false,
		public languageName: string = '',
		public languageTag: string = '',
		public isPublic: boolean = false,
		public searchIndexingEnabled?: boolean,
		public imageCount: number = 0,
	) {}
}

export class GoalPostHeader {
	constructor(
		public goalId: number = 0,
		/** Unix epoch time seconds */
		public dateTime: number = 0,
		public isPublic: boolean = false,
		/** "post" or "comment" */
		public type: string = '',
		public title: string = '',
		public googlePingedAt?: number,
		public publicUrl?: string,
	) {}

	get yearAndMonthText(): string {
		return DateTime.fromMillis(this.dateTime * 1000).toFormat('yyyy-MM');
	}

	get dateText(): string {
		return DateTime.fromMillis(this.dateTime * 1000).toFormat('yyyy-MM-dd');
	}
}
